ui/Modal/Org/ConfigureEns/RegisterName.svelte

```diff
@@ -0,0 +1,289 @@
+<!--
+ Copyright © 2021 The Radicle Upstream Contributors
+
+ This file is part of radicle-upstream, distributed under the GPLv3
+ with Radicle Linking Exception. For full terms see the included
+ LICENSE file.
+-->
+<script lang="typescript" context="module">
+  export interface Result {
+    registration: ensResolver.Registration | null;
+    name: string;
+  }
+</script>
+
+<script lang="typescript">
+  import { debounce } from "lodash";
+  import * as ethers from "ethers";
+
+  import { unreachable } from "ui/src/unreachable";
+  import * as ensRegistrar from "ui/src/org/ensRegistrar";
+  import * as ensResolver from "ui/src/org/ensResolver";
+  import * as error from "ui/src/error";
+  import * as mutexExecutor from "ui/src/mutexExecutor";
+  import * as notification from "ui/src/notification";
+  import * as svelteStore from "ui/src/svelteStore";
+  import * as transaction from "ui/src/transaction";
+  import * as validation from "ui/src/validation";
+  import * as wallet from "ui/src/wallet";
+
+  import { Modal, TextInput } from "ui/DesignSystem";
+
+  import ButtonRow from "./ButtonRow.svelte";
+  import ConfirmRegistration from "./ConfirmRegistration.svelte";
+
+  export let registrationDone: (result: Result) => void;
+  export let currentName: string | undefined;
+  export let fee: ethers.BigNumber;
+
+  type State =
+    | {
+        type: "validateAndCommit";
+      }
+    | {
+        type: "register";
+        commitmentSalt: Uint8Array;
+        commitmentBlock: number;
+        minimumCommitmentAge: number;
+      };
+
+  let state: State = { type: "validateAndCommit" };
+  let nameInputValue: string = currentName || "";
+
+  let userInputStarted: boolean = nameInputValue !== "";
+  let commitInProgress: boolean = false;
+
+  let validatedName: string;
+  let validationState: validation.ValidationState = {
+    status: validation.ValidationStatus.NotStarted,
+  };
+  let registration: ensResolver.Registration | null;
+
+  const validateFormExecutor = mutexExecutor.create();
+  const debouncedValidateFormAndSetState = debounce(
+    validateFormAndSetState,
+    1000 // 1 second
+  );
+
+  $: runAllValidations(nameInputValue);
+
+  function runAllValidations(name: string | undefined): void {
+    if (!userInputStarted) {
+      userInputStarted = true;
+      return;
+    }
+
+    if (!name) {
```

Comment:
Could you flip the if-else-statement?

Reply:
be3f7079135357c303d8ba97ba29652324583f54